src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,408 @@
+            SparkMaxSettings shooterSettings = SparkMaxSettingsBuilder.defaults().withCurrentLimits(SHOOTER_MOTOR.CURRENT_LIMIT)
+                    .coastInNeutral().withDirections(false, false).limitMotorOutputs(SHOOTER_MOTOR.SHOOTER_MAXOUTPUT, SHOOTER_MOTOR.SHOOTER_MINOUTPUT)
+                    .withMotorRampUpOnStart(SHOOTER_MOTOR.SHOOTER_MOTOR_RAMPUP).useSmartMotionControl()
+                    .withPositionGains(SHOOTER_MOTOR.SHOOTER_PID_F, SHOOTER_MOTOR.SHOOTER_PID_P, SHOOTER_MOTOR.SHOOTER_PID_I, SHOOTER_MOTOR.SHOOTER_PID_D)
+                    .useAccelerationStrategy(AccelStrategy.kSCurve).withMaxVelocity(SHOOTER_MOTOR.SHOOTER_MAX_RPM).withMaxAcceleration(SHOOTER_MOTOR.SHOOTER_MAX_ACCEL).withClosedLoopError(SHOOTER_MOTOR.SHOOTER_TOLERANCE).build();
+            shooterMotorController = new SparkSpeedController(shootMotor, shooterSettings);
+            shooterMotorController.configure();
+
+        }
+        if (hoodMotorMounted) {
+            hoodMotor = new WPI_TalonSRX(RobotConstants.CAN.HOOD_MOTOR);
+            TalonSettings hoodSettings = TalonSettingsBuilder.defaults().withCurrentLimits(1, 1, 1).brakeInNeutral()
+                    .withDirections(false, false).noMotorOutputLimits().noMotorStartupRamping().usePositionControl()
+                    .withGains(HOOD_MOTOR.HOOD_PID_F, HOOD_MOTOR.HOOD_PID_P, HOOD_MOTOR.HOOD_PID_I, HOOD_MOTOR.HOOD_PID_D)
+                    .withMotionProfile(HOOD_MOTOR.HOOD_CRUISE_VELOCITY, HOOD_MOTOR.HOOD_ACCELERATION, HOOD_MOTOR.ALLOWABLE_ERROR).build();
+
+            hoodMotorController = new TalonPositionController(hoodMotor, hoodSettings);
+            hoodMotorController.configure();
+            hoodMotor.configForwardLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen,
+                    0);
+            hoodMotor.configReverseLimitSwitchSource(LimitSwitchSource.FeedbackConnector, LimitSwitchNormal.NormallyOpen,
+                    0);
+
+            hoodMotor.overrideLimitSwitchesEnable(true);
+            HOOD_POSITION = hoodMotorController.getActualPosition();
+        }
+    }
+
+    //Current structure of shooter is in auto it will be dictated purely by vision
+    //in manual it will be adjusted by alex and with the presets
+    @Override
+    public void customPeriodic(RobotPose rPose, FieldPose fPose) {
+        logging(rPose);
+        logger.log("Current command", getCurrentCommand());
+        ShooterConfiguration config;
+        logger.log("Shooter is on", shootOn);
+        logger.log("Hood current position1", hoodMotorController.getActualPosition());
+        logger.log("Hood Desired Position1", hoodMotorController.getDesiredPosition());
+        logger.log("Preset on ", preset2);
+        logger.log("SHOOT", atShootSpeed() && atHoodPosition());
+        logger.log("Shooter at speed", atShootSpeed());
+        logger.log("Hood at pose", atHoodPosition());
+        if(shootOn){
```

Comment:
We probably need to move all this into command logic.  Again can we limit Periodic routines to logging only.